### ISSUE TITLE
Account API

### ISSUE THREAD

**Issue by a3020** [a3020/concrete5-ideas#7]
Ideally:
- Every account on concrete5.org could use an API
- Via the API, every account can read account related data

Data that would be relevant to request could consist of:
- Package data
- Sales data
- Support data

For packages it would be handy to also get a few statistics, e.g. number of downloads, number of installations.

@ramonleenders @mnakalay would you rather have access to an API, or a webhook? Personally I'd prefer an API, because it's more powerful. 

Related to https://github.com/a3020/concrete5-ideas/pull/3 and https://github.com/a3020/concrete5-ideas/pull/5

**Comment by a3020** [comment]
@mnakalay I guess the reason I'd prefer an API is that with webhooks you can't retrieve information from the past. You'd only receive it from the moment you set up an endpoint.

**Comment by mnakalay** [comment]
An API would have advantages but then the responsibility to check for new sales would be on us.

To be honest, I favor the webhooks because I am not sure the core team would go for an API.

Like I said they expressed very clearly that they have very limited bandwidth and can only do so much. So an API seems like a lot to ask.

Now if they went for it then, by all means, I'm all for it.

**Comment by ramonleenders** [comment]
Ideally, both would be nice. But how do you retrieve package data with the webhook then? Let's say I want to know visitation (analytics) of pages of Add-On [xxx]. It's not that the webhook will push that every time. How would this work exactly? 

With sales it could be darn useful to have a webhook, since I'm now manually adding them in my system based on the emails (contents) I get. Would save me quite the work if this can be done with a webhook.

**Comment by a3020** [comment]
A webhook for sales sounds fun, but e.g. what should happen if the endpoint is down? Or if it gives a server error? Should C5 then retry endlessly? Or do we risk missing data? It's just not ideal, I think. With an API you'd know for sure and you'll have access to data from the past.

**Comment by ramonleenders** [comment]
True, but just fire the webhook once and afterwards you can still use the API. That's why I said both would be nice, since if you only use the API, you don't know the very second you have a sale/refund/whatever. With webhooks, you will always be up-to-date (if your server is up and running that is).

**Comment by a3020** [comment]
OK, I think it'd make sense to make two proposals, one for the API, and one for the webhook. Then C5 can decide.
